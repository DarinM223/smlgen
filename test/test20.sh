../smlgen test20.sml no:s --recurmod --test
if cmp --silent test20.sml.test test20.sml.expected; then
  echo "test20 succeeded"
else
  echo "test20 failed"
  echo "Diff:"
  diff test20.sml.test test20.sml.expected
  exit 1
fi