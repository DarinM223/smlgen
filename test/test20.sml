structure A =
struct
  datatype 'a foo = Foo of 'a B.t
  type 'b B_t = 'b foo
  datatype 'a t = T of 'a B_t
end
structure B =
struct
  datatype 'a bar = Bar of 'a A.t
  type 'b A_t = 'b bar
  datatype 'a t = T of 'a A_t
end
