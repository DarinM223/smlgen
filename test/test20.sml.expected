structure Component1 =
struct
  datatype 'a B_t =
    B_t_T of 'a bar
  and 'a bar =
    Bar of 'a A_t
  and 'a A_t =
    A_t_T of 'a foo
  and 'a foo =
    Foo of 'a B_t
end
structure A =
struct
  datatype foo = datatype Component1.foo
  type 'b B_t = 'b foo
  datatype t = datatype Component1.A_t

end
structure B =
struct
  datatype bar = datatype Component1.bar
  type 'b A_t = 'b bar
  datatype t = datatype Component1.B_t

end